### PR TITLE
Updates for flowcontrol

### PR DESCRIPTION
**Pull request by michaelbausor**
Updates to Pubsub and Logging to support integration of FlowControl into Bundling.

Note that these are changes in the Pubsub surface, in particular in Publisher.java

### CODE REVIEW

**Comment by garrettjonesgoogle** [comment]
I think this improves the surface.

**Comment by pongad** [comment]
The surface change LGTM. This seems to be failing on Travis though.

File: pom.xml [hunk 1]
```diff
@@ -92,7 +92,7 @@
     <github.global.server>github</github.global.server>
     <google.auth.version>0.6.0</google.auth.version>
     <grpc.version>1.0.3</grpc.version>
-    <gax.version>0.1.3</gax.version>
+    <gax.version>0.1.5</gax.version>
```
Comment (garrettjonesgoogle) [inline pom.xml]: Since the bundling stuff is breaking, could we release the next gax at 0.2.0 instead?

**Comment by michaelbausor** [comment]
Updated with generated surface refresh, PTAL. Note that this will continue to fail CI because the relevant GAX version has not been released. Once we have stabilized the changes to GAX, we can merge and release the new version, and then rerun CI here.

File: pom.xml (hunk 1 above)
Comment (michaelbausor) [inline pom.xml]: Yes will do

**Comment by garrettjonesgoogle** [comment]
LGTM, but let's wait until https://github.com/GoogleCloudPlatform/google-cloud-java/pull/1678 is merged before merging this one.

**Comment by shinfan** [comment]
#1678 is merged!

**Pull request merged**